packages/artifact/src/internal/shared/util.ts

```diff
@@ -69,3 +70,78 @@ export function getBackendIdsFromToken(): BackendIds {
 
   throw InvalidJwtError
 }
+
+/**
+ * Masks the `sig` parameter in a URL and sets it as a secret.
+ *
+ * @param url - The URL containing the signature parameter to mask
+ * @remarks
+ * This function attempts to parse the provided URL and identify the 'sig' query parameter.
+ * If found, it registers both the raw and URL-encoded signature values as secrets using
+ * the Actions `setSecret` API, which prevents them from being displayed in logs.
+ *
+ * The function handles errors gracefully if URL parsing fails, logging them as debug messages.
+ *
+ * @example
+ * ```typescript
+ * // Mask a signature in an Azure SAS token URL
+ * maskSigUrl('https://example.blob.core.windows.net/container/file.txt?sig=abc123&se=2023-01-01');
+ * ```
+ */
+export function maskSigUrl(url: string): void {
+  if (!url) return
+  try {
+    const parsedUrl = new URL(url)
+    const signature = parsedUrl.searchParams.get('sig')
+
+    if (signature) {
+      setSecret(signature)
+      setSecret(encodeURIComponent(signature))
+      parsedUrl.searchParams.set('sig', '***')
```

Comment:
```suggestion
```

This reassignment is ineffectual

Reply:
ah ya just realised that myself, also I removed the encodeURIComponent since I don't think it actually does anything, since we are expecting an encoded URL